lib/src/lib/dataset.ts

```diff
@@ -0,0 +1,170 @@
+    if (ignoredFiles.length > 1) {
+      throw new Error(
+        "More than one file has the ignored field set to true. " +
+        "Only one template/boilerplate code file is allowed at this moment."
+      );
+    }
+    else if (ignore) {
+      return (await readPath(ignore)).ok();
+    }
+    return ignoredFiles.length === 1 ? ignoredFiles[0] : undefined;
+  }
+
+
+  private async fromZIP(
+    zipPath: string,
+    ignore?: string
+  ): Promise<Result<{ nonIgnoredFiles: File[], ignoredFile: File | undefined }>> {
+    const tmpDir = await fs.mkdtemp(path.join(tmpdir(), "dolos-unzip-"));
+    try {
+      const { status, error, stderr } = spawn("unzip", [zipPath, "-d", tmpDir]);
+      if (error) {
+        throw error;
+      } else if (status != 0) {
+        throw new Error(`Unzipping failed with exit status ${ status }, stderr: \n${stderr}`);
+      }
+      const infoPath = path.join(tmpDir, "info.csv");
+      if (await fs.access(infoPath, constants.R_OK).then(() => true).catch(() => false)) {
+        const { nonIgnoredFiles, ignoredFile } = (await this.fromCSV(infoPath, ignore)).ok();
+        if (nonIgnoredFiles) {
+          return Result.ok({ nonIgnoredFiles, ignoredFile });
+        }
+        else {
+          throw new Error("Failed to process files");
+        }
+      } else {
+        const nonIgnoredFiles = (await this.fromDirectory(tmpDir)).ok();
+        const ignoredFile = undefined;
+        return Result.ok({ nonIgnoredFiles, ignoredFile });
+      }
+    } finally {
+      await fs.rm(tmpDir, { recursive: true });
+    }
+  }
+
+
+  private async fromCSV(
+    infoPath: string,
+    ignore?: string
+  ): Promise<Result<{ nonIgnoredFiles: File[], ignoredFile: File | undefined }>> {
+    const dirname = path.dirname(infoPath);
+    try {
+      const files = csvParse((await fs.readFile(infoPath)).toString())
+        .map((row:  DSVRowString) => ({
+          filename: row.filename as string,
+          fullName: row.full_name as string,
+          id: row.id as string,
+          status: row.status as string,
+          submissionID: row.submission_id as string,
+          nameEN: row.name_en as string,
+          nameNL: row.name_nl as string,
+          exerciseID: row.exercise_id as string,
+          createdAt: new Date(row.created_at as string),
+          labels: row.label as string || row.labels as string,
+          ignored: row.ignored as string
+        }))
+        .map((row: ExtraInfo) => readPath(path.join(dirname, row.filename), row));
+      const resolvedFiles = await Result.all(files);
+      const ignoredFile = await this.setIgnoredFile(resolvedFiles.ok(), ignore);
+      const nonIgnoredFiles = resolvedFiles.ok().filter(file => file.extra?.ignored !== "true");
+      return Result.ok({ nonIgnoredFiles, ignoredFile });
+    } catch(e) {
+      throw new Error("The given '.csv'-file could not be opened");
+    }
+  }
+
+
+  public async createNameCandidate(paths: string[]): Promise<string> {
```

Comment:
Now that it is written out, I don't think a separate function to decide the name candidate is a good idea, because we do the same checks to see which situation we are in.

If we would need to support other checks in the future, this would require adding a case to each of these checks.